xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -3528,6 +3528,13 @@ bool CVideoPlayer::OpenStream(CCurrentStream& current, int64_t demuxerId, int iS
     if(!stream || stream->disabled)
       return false;
 
+    if (current.type == STREAM_VIDEO)
```

Comment:
you have to admit that this is rather ugly. it won't solve the problem either because player can call SetVideoResolution at any time.